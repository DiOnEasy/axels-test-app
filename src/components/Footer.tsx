import React, { FC } from 'react';

import { StyledFooter } from 'styled/StyledFooter';

export const Footer: FC = () => {
    const currentYear = new Date().getFullYear();
    return <StyledFooter>&copy; 2018-{currentYear}</StyledFooter>;
};
