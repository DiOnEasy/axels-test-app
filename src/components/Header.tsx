import React, { FC } from 'react';

import { StyledHeader } from 'styled/StyledHeader';

export const Header: FC = () => <StyledHeader>Test app</StyledHeader>;
