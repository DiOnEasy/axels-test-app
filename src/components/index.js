export { Footer } from 'components/footer/Footer';
export { Header } from 'components/header/Header';
export { GalleryImage } from 'components/galleryPage/galleryImage/GalleryImage';
export { ImageModal } from 'components/galleryPage/imageModal/ImageModal';
export { ModalButton } from 'components/galleryPage/modalButton/ModalButton';
export { ModalInput } from 'components/galleryPage/modalInput/ModalInput';
export { CommentForm } from 'components/galleryPage/commentForm/CommentForm';
