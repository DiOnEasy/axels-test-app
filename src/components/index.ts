export { Footer } from 'components/Footer';
export { Header } from 'components/Header';
export { GalleryImage } from 'components/galleryPage/GalleryImage';
export { ImageModal } from 'components/galleryPage/ImageModal';
export { ModalButton } from 'components/galleryPage/ModalButton';
export { ModalInput } from 'components/galleryPage/ModalInput';
export { CommentForm } from 'components/galleryPage/CommentForm';
