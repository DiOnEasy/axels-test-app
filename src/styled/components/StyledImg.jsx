import styled from "styled-components";

export const StyledImg = styled.img.attrs((props) =>({
  src:  props.src || null,
}))`
  width: 100%;
`;
