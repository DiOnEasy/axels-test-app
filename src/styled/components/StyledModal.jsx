import styled from "styled-components";

export const ModalWrapper = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  padding: 25px;
  background: #ffffff;
  width: 60%;
  min-width: 650px;
`;
