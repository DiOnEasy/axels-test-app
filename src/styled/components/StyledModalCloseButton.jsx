import styled from "styled-components";

export const StyledModalCloseButton = styled.div`
  position: absolute;
  right: 0;
  top: 0;
  &::before,
  &::after {
    content: "";
    position: absolute;
    top: 50%;
    left: 50%;
    width: 100%;
    height: 2px;
    background-color: black;
  }
`;
