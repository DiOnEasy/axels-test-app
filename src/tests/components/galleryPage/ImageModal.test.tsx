import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';

import { ImageModal } from 'components';

import configureStore from 'redux-mock-store';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useParams: () => ({ id: '1' })
}));

const mockStore = configureStore([]);

const store = mockStore({
    images: {
        imageDetails: {
            id: '1',
            url: 'https://example.com/image.jpg',
            comments: [
                { name: 'claus', date: 3123782132, text: 'Nice picture!' }
            ]
        },
        imageDetailsLoading: false
    }
});

describe('ImageModal Component', () => {
    const renderComponent = () => {
        return render(
            <Provider store={store}>
                <ImageModal open={true} onCloseImageModal={() => {}} />
            </Provider>
        );
    };

    it('should render img', () => {
        renderComponent();
        expect(screen.getByRole('img')).toBeInTheDocument();
    });

    it('should render comment date', () => {
        renderComponent();
        expect(screen.getByText('26.12.2068')).toBeInTheDocument();
    });

    it('should render comment text', () => {
        renderComponent();
        expect(screen.getByText('Nice picture!')).toBeInTheDocument();
    });

    it('should render two paragraph elements', () => {
        renderComponent();
        expect(screen.getAllByRole('paragraph')).toHaveLength(2);
    });

    it('should match the snapshot when open is true', () => {
        const { baseElement } = render(
            <Provider store={store}>
                <ImageModal open={true} onCloseImageModal={() => {}} />
            </Provider>
        );

        expect(baseElement).toMatchSnapshot();
    });
});
