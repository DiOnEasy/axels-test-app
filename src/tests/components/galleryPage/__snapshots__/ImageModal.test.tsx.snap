// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageModal Snapshot renders correctly 1`] = `<DocumentFragment />`;
